rotkehlchen/chain/bitcoin/bch/manager.py

```diff
@@ -30,10 +35,27 @@ def __init__(self, database: 'DBHandler') -> None:
                 name='haskoin',
                 balances_fn=self._query_haskoin_balances,
                 has_transactions_fn=self._query_haskoin_has_transactions,
-                transactions_fn=None,  # TODO: add support for querying transactions
+                transactions_fn=self._query_haskoin_transactions,
             )],  # TODO: add other APIs for fallbacks if haskoin goes down
         )
 
+    def refresh_tracked_accounts(self) -> None:
+        """Refresh the tracked accounts and ensure the CashAddr format of each address is used
+        since all addresses returned from haskoin are in the CashAddr format.
+        """
+        super().refresh_tracked_accounts()
+        converted_addresses = {}
+        for account in self.tracked_accounts:
+            if (
+                not is_valid_bitcoin_cash_address(account) and
+                (addr := legacy_to_cash_address(account)) is not None
+            ):
+                converted_addresses[account] = addr
+
+        for legacy_addr, cash_addr in converted_addresses.items():
+            self.tracked_accounts.remove(legacy_addr)
+            self.tracked_accounts.append(cash_addr)
```

Comment:
```suggestion
    super().refresh_tracked_accounts()
    
    for i, account in enumerate(self.tracked_accounts):
        if not is_valid_bitcoin_cash_address(account):
            cash_addr = legacy_to_cash_address(account)
            if cash_addr:
                self.tracked_accounts[i] = cash_addr
```

or you can just create a conversion cache attribute in the class instance. since you'd need to know which cashaddr matches a legacy address for users that added legacy addresses.